OpenRA.Mods.Common/Traits/Rearmable.cs

```diff
@@ -44,17 +44,14 @@ void INotifyCreated.Created(Actor self)
 			RearmableAmmoPools = self.TraitsImplementing<AmmoPool>().Where(p => Info.AmmoPools.Contains(p.Info.Name)).ToArray();
 		}
 
-		void INotifyResupply.BeforeResupply(Actor self, Actor target, ResupplyType types)
+		void INotifyDockClient.Docked(Actor self, Actor dock)
 		{
-			if (!types.HasFlag(ResupplyType.Rearm))
-				return;
```

Comment:
Removing the type filtering here feels like a behaviour change!

Reply:
Oh my bad. In this specific case the code isn't even functional. See the commit name.

Reply:
it unfunctional because it uses a trait that supposed to be used on the host actor, while in reality it exists on the client